src/main/java/net/sacredlabyrinth/phaed/dynmap/simpleclans/DynmapSimpleClans.java

```diff
@@ -1,181 +1,158 @@
+     *
+     * @param message             message to send
+     * @param respectUserDecision should the message be sent if debug is false?
+     */
+    public static void debug(String message, boolean respectUserDecision) {
+        if (respectUserDecision && !instance.getConfig().getBoolean("debug", false)) {
+            return;
+        }
+
+        instance.getLogger().info(message);
+    }
+
+    public static void debug(String message) {
+        debug(message, true);
+    }
+
+    @Override
+    public void onEnable() {
+        instance = this;
+
+        reload();
+
+        CommandManager commandManager = new CommandManager(this);
+        PluginCommand clanmap = Objects.requireNonNull(getCommand("clanmap"));
+        clanmap.setExecutor(commandManager);
+        clanmap.setTabCompleter(commandManager);
+
+        getPluginManager().registerEvents(new DynmapSimpleClansListener(this), this);
+
+        getLogger().log(Level.INFO, "Version {0} is activated!", getDescription().getVersion());
+    }
+
+    public void reload() {
+        loadPreferences();
+        loadDependencies();
+        loadTasks();
+        saveDefaultConfig();
+        saveDefaultImages();
+        loadLayers();
+    }
+
+    public void loadLayers() {
+        ConfigurationSection clanHomesSection = Objects.requireNonNull(getConfig().getConfigurationSection("layer.homes"));
+        ConfigurationSection killsSection = Objects.requireNonNull(getConfig().getConfigurationSection("layer.kills"));
+
+        String defaultIconName = clanHomesSection.getString("default-icon", "clanhome");
+
+        IconStorage homeIcons = new IconStorage(this, "/images/clanhome", defaultIconName, markerApi);
+        IconStorage killsIcons = new IconStorage(this, "/images", "blood", markerApi);
+
+        try {
+            homeLayer = new HomeLayer(homeIcons, new LayerConfig(clanHomesSection), markerApi);
+            killsLayer = new KillsLayer(killsIcons, new LayerConfig(killsSection), markerApi);
+        } catch (IllegalStateException ex) {
+            debug(ex.getMessage());
+        }
+    }
+
+    public ClanManager getClanManager() {
+        return simpleclans.getClanManager();
+    }
+
+    public DynmapAPI getDynmapApi() {
+        return dynmapApi;
+    }
+
+    @Nullable
+    public HomeLayer getHomeLayer() {
+        return homeLayer;
+    }
+
+    @Nullable
+    public KillsLayer getKillsLayer() {
+        return killsLayer;
+    }
+
+
+    private void loadDependencies() {
+        dynmapApi = (DynmapAPI) getServer().getPluginManager().getPlugin("DynMap");
+        if (dynmapApi != null) {
+            markerApi = dynmapApi.getMarkerAPI();
+            if (markerApi == null) {
+                getLogger().severe("'markers' component has not been configured in DynMap!");
+            }
+        }
+
+        simpleclans = SimpleClans.getInstance();
+    }
+
+    private void saveDefaultImages() {
+        if (!new File(getDataFolder(), "/images/clanhome").exists()) {
```

Comment:
No extension.
Also, what if I want to disable the clanhome icon? It will be back after every reload...